mx.c

```diff
@@ -1685,6 +1685,77 @@ struct Mailbox *mx_path_resolve(const char *path)
   return m;
 }
 
+/**
+ * mx_mbox_find_by_name - Find a Mailbox with given name under an account.
+ * @param a    Account to search
+ * @param name Name to find
+ * @retval ptr Mailbox
+ */
+static struct Mailbox *mx_mbox_find_by_name_ac(struct Account *a, const char *name)
+{
+  if (!a || !name)
+    return NULL;
+
+  struct MailboxNode *np = NULL;
+
+  STAILQ_FOREACH(np, &a->mailboxes, entries)
+  {
+    if (mutt_str_equal(np->mailbox->name, name))
+      return np->mailbox;
+  }
+
+  return NULL;
+}
+
+/**
+ * mx_mbox_find_by_name - Find a Mailbox with given name.
+ * @param name Name to search
+ * @retval ptr Mailbox
+ */
+static struct Mailbox *mx_mbox_find_by_name(const char *name)
+{
+  if (!name)
+    return NULL;
+
+  char buf[PATH_MAX];
+  mutt_str_copy(buf, name, sizeof(buf));
```

Comment:
Why the additional buf instead of just passing name to mx_mbox_find_by_name_ac?

Reply:
No reason. That was just an artifact leftover from basing the function on `mx_mbox_find2()`